server/clients/prisonApiClient.js

```diff
@@ -0,0 +1,99 @@
+const { path, pathOr } = require('ramda');
+const { baseClient } = require('./baseClient');
+const { logger } = require('../utils/logger');
+const { InMemoryCachingStrategy } = require('../utils/caching');
+
+const PRISON_API_TOKEN_KEY = 'prisonApi:bearerToken';
+const CACHE_EXPIRY_OFFSET = 60; // Seconds
+
+class PrisonApiClient {
+  constructor(options = {}) {
+    const { clientId, clientSecret, authUrl } = pathOr(
+      {},
+      ['prisonApi', 'auth'],
+      options,
+    );
+    this.client = options.client || baseClient;
+    this.cache = options.cachingStrategy || new InMemoryCachingStrategy();
+    this.setBasicAuthToken(clientId, clientSecret);
+
+    if (!authUrl) {
```

Comment:
I'd move this before doing any of the `this` stuff purely as it could call two functions unnecessarily if `authUrl` is not supplied. Fail fast and all that?

Reply:
Moved to fail fast